.github/workflows/iroha2-dev-lints.yml

```diff
@@ -0,0 +1,111 @@
+name: I2::Dev::Lints
+
+on:
+  workflow_run:
+    workflows: ["I2::Dev::Tests", "I2::Dev::Static"]
+    types: [requested]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  workspace_analysis_clippy:
+    runs-on: ubuntu-latest
+    container:
+      image: hyperledger/iroha2-ci:nightly-2024-04-18
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+      - name: Format
+        run: cargo fmt --all -- --check
+      - name: Lints without features
+        if: always()
+        run: cargo clippy --workspace --benches --tests --examples --no-default-features --quiet
+      - name: Lints with all features enabled
+        if: always()
+        run: cargo clippy --workspace --benches --tests --examples --all-features --quiet --message-format=json | tee clippy.json
+      - name: Documentation
+        if: always()
+        run: cargo doc --no-deps --quiet
+      - name: Upload clippy report artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: clippy.json
+          path: clippy.json
+
+  # exclude: client/tests/integration/
+  with_coverage:
+    runs-on: [self-hosted, Linux, iroha2]
+    container:
+      image: hyperledger/iroha2-ci:nightly-2024-04-18
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+      - name: Run tests, with coverage
+        run: |
+          mold --run cargo test --all-features --no-fail-fast --workspace --exclude iroha_client
+          mold --run cargo test --all-features --no-fail-fast -p iroha_client -- --skip integration
+        env:
+          RUSTFLAGS: "-C instrument-coverage"
+          LLVM_PROFILE_FILE: "iroha-%p-%m.profraw"
+      - name: Generate lcov report
+        if: always()
+        run: grcov . --binary-path ./target/debug/ -s . -t lcov --branch --ignore-not-existing --ignore "/client_cli" --ignore "**/main.rs" -o lcov.info
+      - name: Upload coverage to coveralls.io
```

Comment:
One thing more: within this workflow trigger, I'm not sure if coveralls result will be appear in PR checks among with PR comment by coveralls

Reply:
as long as the job fails and we get a link to the report it's fine